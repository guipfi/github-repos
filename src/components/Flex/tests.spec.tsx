import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import { Flex } from '.';

describe('flex tests', () => {
  it('should render a column based flex container', () => {
    const component = renderer.create(
      <Flex>
        <Text>First line</Text>
        <Text>Second line</Text>
      </Flex>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a row based flex container', () => {
    const component = renderer.create(
      <Flex style={{ flexDirection: 'row' }}>
        <Text>First column</Text>
        <Text>Second column</Text>
      </Flex>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
