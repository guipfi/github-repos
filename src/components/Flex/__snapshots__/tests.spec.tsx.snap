// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flex tests should render a column based flex container 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <Text>
    First line
  </Text>
  <Text>
    Second line
  </Text>
</View>
`;

exports[`flex tests should render a row based flex container 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "column",
      },
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Text>
    First column
  </Text>
  <Text>
    Second column
  </Text>
</View>
`;
